src/crypto/openssl.c

```diff
@@ -534,9 +672,398 @@ SSL_CTX *create_ssl_context_from_options(struct us_socket_context_options_t opti
+        if (SSL_CTX_load_verify_locations(ssl_context, options.ca_file_name, NULL) != 1) {
+            free_ssl_context(ssl_context);
+            return NULL;
+        }
+        
+        if(options.reject_unauthorized) {
+            SSL_CTX_set_verify(ssl_context, SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, us_verify_callback);
+        } else {
+            SSL_CTX_set_verify(ssl_context, SSL_VERIFY_PEER, us_verify_callback);
+        }
+
+     }else if (options.ca && options.ca_count > 0) {
+        X509_STORE* cert_store = NULL;
+        
+        for(unsigned int i = 0; i < options.ca_count; i++){
+            X509* ca_cert = us_ssl_ctx_get_X509_from(ssl_context, options.ca[i]);
+            if (ca_cert == NULL){
+                free_ssl_context(ssl_context);
+                return NULL;
+            }
+
+            if (cert_store == NULL) {
+                cert_store = X509_STORE_new();
+                X509_STORE_set_default_paths(cert_store);
+                SSL_CTX_set_cert_store(ssl_context, cert_store);
+            }
+    
+            X509_STORE_add_cert(cert_store, ca_cert);
+            if(!SSL_CTX_add_client_CA(ssl_context, ca_cert)){
+                free_ssl_context(ssl_context);
+                return NULL;
+            }
+            if(options.reject_unauthorized) {
+                SSL_CTX_set_verify(ssl_context, SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, us_verify_callback);
+            } else {
+                SSL_CTX_set_verify(ssl_context, SSL_VERIFY_PEER, us_verify_callback);
+            }
+        }
+    } else {
+        if(options.request_cert) {
+            if(options.reject_unauthorized) {
+                SSL_CTX_set_verify(ssl_context, SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, us_verify_callback);
+            } else {
+                SSL_CTX_set_verify(ssl_context, SSL_VERIFY_PEER, us_verify_callback);
+            }
+        }
+    }
+    if (options.dh_params_file_name) {
+        /* Set up ephemeral DH parameters. */
+        DH *dh_2048 = NULL;
+        FILE *paramfile;
+        paramfile = fopen(options.dh_params_file_name, "r");
+
+        if (paramfile) {
+            dh_2048 = PEM_read_DHparams(paramfile, NULL, NULL, NULL);
+            fclose(paramfile);
+        } else {
+            free_ssl_context(ssl_context);
+            return NULL;
+        }
+
+        if (dh_2048 == NULL) {
+            free_ssl_context(ssl_context);
+            return NULL;
+        }
+
+        const long set_tmp_dh = SSL_CTX_set_tmp_dh(ssl_context, dh_2048);
+        DH_free(dh_2048);
+
+        if (set_tmp_dh != 1) {
+            free_ssl_context(ssl_context);
+            return NULL;
+        }
+
+        /* OWASP Cipher String 'A+' (https://www.owasp.org/index.php/TLS_Cipher_String_Cheat_Sheet) */
+        if (SSL_CTX_set_cipher_list(ssl_context, "DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256") != 1) {
```

Comment:
is this supposed to go in this scope? or is it supposed to be set when `options.SSL_ciphers` is not set

Reply:
Actually that's no need to set this cyphers manually. I will update this.

Reply:
Taking a deeper look is actually fine, will be only the default if no ciphers are informed and only when dh_params_file_name is informed, we can revisit it when we add dh_params (in memory) support.